ktor-features/ktor-jackson/jvm/src/io/ktor/jackson/JacksonConverter.kt

```diff
@@ -55,12 +56,17 @@ public class JacksonConverter(private val objectmapper: ObjectMapper = jacksonOb
         val type = request.typeInfo
         val value = request.value as? ByteReadChannel ?: return null
         return withContext(Dispatchers.IO) {
-            val reader = value.toInputStream().reader(context.call.request.contentCharset() ?: Charsets.UTF_8)
-            objectmapper.readValue(reader, type.jvmErasure.javaObjectType)
+            try {
+                val reader = value.toInputStream().reader(context.call.request.contentCharset() ?: Charsets.UTF_8)
+                objectmapper.readValue(reader, type.jvmErasure.javaObjectType)
+            } catch (e: JsonMappingException) {
+                throw JsonTransformationException("error parsing json request body: ${e.pathReference}")
```

Comment:
What do you suggest? I thought that the path reference contains everything that's needed to track down the problem.

Reply:
i could not add the exception cause because I did not want to change any public classes. 

ContentTransformationException does not have a cause: 
https://github.com/ktorio/ktor/blob/79fe29bcb253d105dd6c3acda8426a7600f7469d/ktor-server/ktor-server-core/jvm/src/io/ktor/features/Errors.kt#L67

Reply:
after sleeping over it I think the ContentNegotiation module should define two exceptions, one for malformed data (i.e. malformed json that does not validate, and error 400), and the other one for correct syntax but wrong fieldnames (error  422)